### PR TITLE
feat: Set groups based on OIDC

### PR DESCRIPTION
**Pull request by fionera**
This is a first try at getting oidc groups to work

### CODE REVIEW

File: server/modules/authentication/oidc/authentication.js [hunk 1]
```diff
@@ -29,6 +29,15 @@ module.exports = {
               email: _.get(profile, '_json.' + conf.emailClaim)
             }
           })
+          const groups = _.get(profile, '_json.groups')
```
Comment (NGPixel) [inline server/modules/authentication/oidc/authentication.js:32]: Make the `groups` path a configurable property.

**Review by NGPixel — CHANGES_REQUESTED** [review]

File: server/modules/authentication/oidc/authentication.js (hunk 1 above)
Comment (fionera) [reply server/modules/authentication/oidc/authentication.js]: I added a config entry for it :)

**Comment by fionera** [comment]
Ah shoot I just saw that now the Group will be added on every login and they are not deduped. In the WebUI there is the group multiple times now. Also old groups wont be deleted. I will make another PR that removes all groups and then relates to the new ones  or diffs them. Idk yet

**Comment by fionera** [comment]
Could you open an Issue for it? I dont know yet when I have time for it :)

**Comment by NGPixel** [comment]
Fixed in v2.5.2588

**Comment by cvonderstein** [comment]
Hi, I just stumbled upon this and I am wondering, is this implemented now? I am kind of confused by the "Fixed in v2.5.2588" comment, since all I found is v2.5.306. What am I missing here?

**Comment by voc0der** [comment]
I also have this problem still, I don't think it was ever truly resolved.

**Comment by vlad-patoka** [comment]
I implemented a federated Cognito setup, where a Cognito user pool in one AWS account acts as the identity provider (IdP) for a Cognito pool in another account. Wiki.js successfully creates user groups regardless of whether the authentication comes from the local or federated Cognito pool. However, to inject custom claims into the token response, I had to use Cognito extensions - specifically, a Lambda trigger for pre-token generation. Also it required a full refactor of authentication.js to properly handle the integration.

**Pull request merged**